c10/core/WrapDimMinimal.cpp

```diff
@@ -3,10 +3,17 @@
 namespace c10 {
 namespace detail {
 
-int64_t maybe_wrap_dim_slow(
+template int64_t maybe_wrap_dim_slow(
     int64_t dim,
     int64_t dim_post_expr,
-    bool wrap_scalar) {
+    bool wrap_scalar);
+template SymInt maybe_wrap_dim_slow(
+    SymInt dim,
+    SymInt dim_post_expr,
+    bool wrap_scalar);
+
+template <typename T>
+T maybe_wrap_dim_slow(T dim, T dim_post_expr, bool wrap_scalar) {
```

Comment:
I want to point out why sometimes you were able to template, and why sometimes you weren't. C++ will attempt implicit conversions when calling functions. However, if the function is templated, no implicit conversions will happen. It turns out that maybe_wrap_dim_slow has no implicit conversions, which is why this is ok

Reply:
yea, i realized that we have a bunch of implicit conversions from .sizes() calls coming into this function, so i gave up for now